homeassistant/components/sungrow/sensor.py

```diff
@@ -0,0 +1,436 @@
+        icon="mdi:transmission-tower",
+    ),
+    SensorEntityDescription(
+        key="work_state", entity_category=EntityCategory.DIAGNOSTIC
+    ),
+    SensorEntityDescription(
+        key="fault_date", entity_category=EntityCategory.DIAGNOSTIC, icon="mdi:alert"
+    ),
+    SensorEntityDescription(
+        key="fault_code", entity_category=EntityCategory.DIAGNOSTIC, icon="mdi:alert"
+    ),
+    SensorEntityDescription(
+        key="nominal_reactive_output_power",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="impedance_to_ground",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="daily_running_time",
+        device_class=SensorDeviceClass.DURATION,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="country",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:earth",
+    ),
+    SensorEntityDescription(
+        key="monthly_power_yield", entity_registry_enabled_default=False
+    ),
+    SensorEntityDescription(
+        key="negative_voltage_to_ground",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="bus_voltage",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_factor_setting",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_adjustment",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_adjustment_switch",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_percentage",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add solarlog entry."""
+    coordinator: SungrowData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SungrowSensor(
+            coordinator,
+            coordinator.battery_info
+            if description.key in BATTERY_DEVICE_VARIABLES
+            else coordinator.device_info,
+            description,
+        )
+        for description in DESCRIPTIONS
+        if description.key in coordinator.client.keys
+    )
+
+
+class SungrowSensor(SungrowCoordinatorEntity, SensorEntity):
```

Comment:
New integrations/entities must support [translation of names and attributes](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/).

Please adopt your code accordingly.